tests/func/test_scm.py

```diff
@@ -61,112 +62,95 @@ def test_commit_in_submodule(self):
-    def _count_gitignore_entries(string):
-        lines = get_gitignore_content()
-        return len([i for i in lines if i == string])
+def _count_gitignore_entries(string):
+    lines = get_gitignore_content()
+    return len([i for i in lines if i == string])
 
-    def test_ignore(self, git, repo_dir):
-        git = Git(repo_dir._root_dir)
-        foo = os.path.join(repo_dir._root_dir, repo_dir.FOO)
 
-        target = "/" + repo_dir.FOO
+def test_ignore(tmp_dir, scm):
+    tmp_dir.gen({"foo": "foo contents"})
 
-        git.ignore(foo)
-        assert os.path.isfile(Git.GITIGNORE)
-        assert self._count_gitignore_entries(target) == 1
+    foo = fspath(tmp_dir / "foo")
+    target = "/foo"
 
-        git.ignore(foo)
-        assert os.path.isfile(Git.GITIGNORE)
-        assert self._count_gitignore_entries(target) == 1
+    scm.ignore(foo)
+    assert os.path.isfile(Git.GITIGNORE)
+    assert _count_gitignore_entries(target) == 1
 
-        git.ignore_remove(foo)
-        assert self._count_gitignore_entries(target) == 0
+    scm.ignore(foo)
+    assert os.path.isfile(Git.GITIGNORE)
+    assert _count_gitignore_entries(target) == 1
 
-    def test_get_gitignore(self, git, repo_dir):
-        data_dir = os.path.join(repo_dir._root_dir, "file1")
-        entry, gitignore = Git(repo_dir._root_dir)._get_gitignore(data_dir)
-        assert entry == "/file1"
-        assert gitignore == os.path.join(repo_dir._root_dir, Git.GITIGNORE)
+    scm.ignore_remove(foo)
+    assert _count_gitignore_entries(target) == 0
 
-        data_dir = os.path.join(repo_dir._root_dir, "dir")
-        entry, gitignore = Git(repo_dir._root_dir)._get_gitignore(data_dir)
 
-        assert entry == "/dir"
-        assert gitignore == os.path.join(repo_dir._root_dir, Git.GITIGNORE)
+def test_get_gitignore(tmp_dir, scm):
+    data_dir = fspath(tmp_dir / "file1")
+    entry, gitignore = scm._get_gitignore(data_dir)
+    assert entry == "/file1"
+    assert gitignore == fspath(tmp_dir / ".gitignore")
 
-    def test_get_gitignore_symlink(self, git, repo_dir):
-        link = os.path.join(repo_dir.root_dir, "link")
-        target = os.path.join(repo_dir.root_dir, repo_dir.DATA_SUB)
-        System.symlink(target, link)
-        entry, gitignore = Git(repo_dir._root_dir)._get_gitignore(link)
-        assert entry == "/link"
-        assert gitignore == os.path.join(repo_dir.root_dir, Git.GITIGNORE)
+    data_dir = fspath(tmp_dir / "dir")
+    entry, gitignore = scm._get_gitignore(data_dir)
 
-    def test_get_gitignore_subdir(self, git, repo_dir):
-        data_dir = os.path.join(
-            repo_dir._root_dir, os.path.join("dir1", "file1")
-        )
-        entry, gitignore = Git(repo_dir._root_dir)._get_gitignore(data_dir)
-        assert entry == "/file1"
-        assert gitignore == os.path.join(
-            repo_dir._root_dir, "dir1", Git.GITIGNORE
-        )
+    assert entry == "/dir"
+    assert gitignore == fspath(tmp_dir / ".gitignore")
```

Comment:
Since there is no file nor dir this is simply duplication.

Reply:
The whole test is checking some private thing, should we do that at all? Maybe ignore this for now.

Reply:
Yep, we should, but in unit tests. Out of focus for this PR though, as I've stated above.